sandbox/Sandbox/Generated.cs

```diff
@@ -2070,19 +1958,176 @@ public void Serialize(ref MessagePackWriter writer, global::SharedData.Callback1
             }
 
             var ____result = new global::SharedData.Callback1_2(__X__);
-            ____result.X = __X__;
-            ((IMessagePackSerializationCallbackReceiver)____result).OnAfterDeserialize();
+            ((global::MessagePack.IMessagePackSerializationCallbackReceiver)____result).OnAfterDeserialize();
+            reader.Depth--;
+            return ____result;
+        }
+    }
+
+    public sealed class DefaultValueIntKeyClassWithExplicitConstructorFormatter : global::MessagePack.Formatters.IMessagePackFormatter<global::SharedData.DefaultValueIntKeyClassWithExplicitConstructor>
+    {
+
+        public void Serialize(ref global::MessagePack.MessagePackWriter writer, global::SharedData.DefaultValueIntKeyClassWithExplicitConstructor value, global::MessagePack.MessagePackSerializerOptions options)
+        {
+            if (value == null)
+            {
+                writer.WriteNil();
+                return;
+            }
+
+            writer.WriteArrayHeader(2);
+            writer.Write(value.Prop1);
+            writer.Write(value.Prop2);
+        }
+
+        public global::SharedData.DefaultValueIntKeyClassWithExplicitConstructor Deserialize(ref global::MessagePack.MessagePackReader reader, global::MessagePack.MessagePackSerializerOptions options)
+        {
+            if (reader.TryReadNil())
+            {
+                return null;
+            }
+
+            options.Security.DepthStep(ref reader);
+            var length = reader.ReadArrayHeader();
+            var __Prop1__ = default(int);
+            var __Prop2__ = default(int);
+
+            for (int i = 0; i < length; i++)
+            {
+                switch (i)
+                {
+                    case 0:
+                        __Prop1__ = reader.ReadInt32();
+                        break;
+                    case 1:
+                        __Prop2__ = reader.ReadInt32();
+                        break;
+                    default:
+                        reader.Skip();
+                        break;
+                }
+            }
+
+            var ____result = new global::SharedData.DefaultValueIntKeyClassWithExplicitConstructor(__Prop1__);
+            if (length <= 1)
+            {
+                goto MEMBER_ASSIGNMENT_END;
+            }
```

Comment:
Presumably this pattern repeats after each property assignment, yes? But all the code generation in this file only has it once since there are only two properties. Can we make this a bit more interesting and demonstrate the loop is correct by adding at least a 3rd property?

Reply:
Okay, 3rd and 4th property is added.